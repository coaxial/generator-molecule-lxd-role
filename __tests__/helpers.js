'use strict';
const assert = require('yeoman-assert');

const { capitalize, choicesFor } = require('../generators/helpers');

describe('helpers', () => {
  describe('#choicesFor', () => {
    it('returns the requested distros only', () => {
      const actual = choicesFor(['ubuntu']);

      expect(actual).toMatchSnapshot();
    });

    it('works with multiple distros', () => {
      const actual = choicesFor(['debian', 'ubuntu']);

      expect(actual).toMatchSnapshot();
    });
  });

  describe('#capitalize', () => {
    it('correctly transforms single words', () => {
      const expected = 'Hello';

      const actual = capitalize('hello');

      assert.equal(actual, expected);
    });
  });
});
