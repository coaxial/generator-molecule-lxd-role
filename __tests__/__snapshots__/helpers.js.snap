// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers #choicesFor returns the requested distros only 1`] = `
Array [
  Separator {
    "line": "[2mâ†“  Ubuntu â†“ [22m",
    "type": "separator",
  },
  Object {
    "name": "All LTS (Ubuntu)",
    "value": Array [
      Object {
        "codeName": "trusty",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "lts",
          "current",
        ],
        "versionNumber": "14.04",
      },
      Object {
        "codeName": "xenial",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "lts",
          "current",
        ],
        "versionNumber": "16.04",
      },
      Object {
        "codeName": "bionic",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "lts",
          "current",
        ],
        "versionNumber": "18.04",
      },
    ],
  },
  Object {
    "name": "All current and future (Ubuntu)",
    "value": Array [
      Object {
        "codeName": "trusty",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "lts",
          "current",
        ],
        "versionNumber": "14.04",
      },
      Object {
        "codeName": "xenial",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "lts",
          "current",
        ],
        "versionNumber": "16.04",
      },
      Object {
        "codeName": "bionic",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "lts",
          "current",
        ],
        "versionNumber": "18.04",
      },
      Object {
        "codeName": "cosmic",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "future",
        ],
        "versionNumber": "18.10",
      },
    ],
  },
  Object {
    "name": "Trusty",
    "value": Object {
      "codeName": "trusty",
      "distribution": "ubuntu",
      "family": "debian",
      "versionNumber": "14.04",
    },
  },
  Object {
    "name": "Xenial",
    "value": Object {
      "codeName": "xenial",
      "distribution": "ubuntu",
      "family": "debian",
      "versionNumber": "16.04",
    },
  },
  Object {
    "name": "Bionic",
    "value": Object {
      "codeName": "bionic",
      "distribution": "ubuntu",
      "family": "debian",
      "versionNumber": "18.04",
    },
  },
  Object {
    "name": "Cosmic",
    "value": Object {
      "codeName": "cosmic",
      "distribution": "ubuntu",
      "family": "debian",
      "versionNumber": "18.10",
    },
  },
]
`;

exports[`helpers #choicesFor works with multiple distros 1`] = `
Array [
  Separator {
    "line": "[2mâ†“  Debian â†“ [22m",
    "type": "separator",
  },
  Object {
    "name": "All LTS (Debian)",
    "value": Array [
      Object {
        "codeName": "stretch",
        "distribution": "debian",
        "family": "debian",
        "tags": Array [
          "lts",
        ],
        "versionNumber": "9",
      },
    ],
  },
  Object {
    "name": "All current and future (Debian)",
    "value": Array [
      Object {
        "codeName": "jessie",
        "distribution": "debian",
        "family": "debian",
        "tags": Array [
          "current",
        ],
        "versionNumber": "8",
      },
      Object {
        "codeName": "buster",
        "distribution": "debian",
        "family": "debian",
        "tags": Array [
          "future",
        ],
        "versionNumber": "testing",
      },
      Object {
        "codeName": "sid",
        "distribution": "debian",
        "family": "debian",
        "tags": Array [
          "future",
        ],
        "versionNumber": "unstable",
      },
    ],
  },
  Object {
    "name": "Jessie",
    "value": Object {
      "codeName": "jessie",
      "distribution": "debian",
      "family": "debian",
      "versionNumber": "8",
    },
  },
  Object {
    "name": "Stretch",
    "value": Object {
      "codeName": "stretch",
      "distribution": "debian",
      "family": "debian",
      "versionNumber": "9",
    },
  },
  Object {
    "name": "Buster",
    "value": Object {
      "codeName": "buster",
      "distribution": "debian",
      "family": "debian",
      "versionNumber": "testing",
    },
  },
  Object {
    "name": "Sid",
    "value": Object {
      "codeName": "sid",
      "distribution": "debian",
      "family": "debian",
      "versionNumber": "unstable",
    },
  },
  Separator {
    "line": "[2mâ†“  Ubuntu â†“ [22m",
    "type": "separator",
  },
  Object {
    "name": "All LTS (Ubuntu)",
    "value": Array [
      Object {
        "codeName": "trusty",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "lts",
          "current",
        ],
        "versionNumber": "14.04",
      },
      Object {
        "codeName": "xenial",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "lts",
          "current",
        ],
        "versionNumber": "16.04",
      },
      Object {
        "codeName": "bionic",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "lts",
          "current",
        ],
        "versionNumber": "18.04",
      },
    ],
  },
  Object {
    "name": "All current and future (Ubuntu)",
    "value": Array [
      Object {
        "codeName": "trusty",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "lts",
          "current",
        ],
        "versionNumber": "14.04",
      },
      Object {
        "codeName": "xenial",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "lts",
          "current",
        ],
        "versionNumber": "16.04",
      },
      Object {
        "codeName": "bionic",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "lts",
          "current",
        ],
        "versionNumber": "18.04",
      },
      Object {
        "codeName": "cosmic",
        "distribution": "ubuntu",
        "family": "debian",
        "tags": Array [
          "future",
        ],
        "versionNumber": "18.10",
      },
    ],
  },
  Object {
    "name": "Trusty",
    "value": Object {
      "codeName": "trusty",
      "distribution": "ubuntu",
      "family": "debian",
      "versionNumber": "14.04",
    },
  },
  Object {
    "name": "Xenial",
    "value": Object {
      "codeName": "xenial",
      "distribution": "ubuntu",
      "family": "debian",
      "versionNumber": "16.04",
    },
  },
  Object {
    "name": "Bionic",
    "value": Object {
      "codeName": "bionic",
      "distribution": "ubuntu",
      "family": "debian",
      "versionNumber": "18.04",
    },
  },
  Object {
    "name": "Cosmic",
    "value": Object {
      "codeName": "cosmic",
      "distribution": "ubuntu",
      "family": "debian",
      "versionNumber": "18.10",
    },
  },
]
`;
